stl/inc/yvals_core.h

```diff
@@ -684,6 +684,10 @@
 #define _HAS_UNEXPECTED (!_HAS_CXX17)
 #endif // _HAS_UNEXPECTED
 
+#if _HAS_UNEXPECTED && _HAS_CXX23
+#error STL1004: C++98 unexpected() is incompatible with C++23 unexpected<E>.

```

Comment:
For posterity: I did verify that STL1000 - STL1003 are all used, so STL1004 is next in sequence. (No change requested.)